packages/core/src/types.ts

```diff
@@ -1814,7 +1817,8 @@ export interface ActorContext<TEvent extends EventObject, TSnapshot> {
 export interface ActorBehavior<
   TEvent extends EventObject,
   TSnapshot = any,
-  TInternalState = any
+  TInternalState = any,
+  TPersisted = TInternalState
```

Comment:
Do we currently have any need for `TPersisted` that is different from `TInternalState`?

Reply:
Yes; in theory, the persisted state may e.g. omit data from the internal state (which you may not want to reveal), or add additional data that the actor behavior knows how to handle when restoring it, like... a checksum or version number or timestamp or something like that.

Reply:
The internal state is hidden from `getSnapshot` but it's likely always (almost?) important for the persisted state. If there is any internal state that can be lost when you persist+rehydrate then u won't get "the same" behavior back

Reply:
OTOH, while I think that we don't need this to "hide" things - we might want this to "serialize" state

Reply:
Made a clarifying comment